In [1]:
import pandas as pd 
df = pd.read_csv('diamonds.csv', sep=',', encoding='utf-8')  

In [27]:
df.describe()

Unnamed: 0         carat         depth         table         price  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
25%        4.710000      4.720000      2.910000  
50%        5.700000      5.710000      3.530000  
75%        6.540000      6.540000      4.040000  
max       10.740000     58.900000     31.800000

El análisis descriptivo de los datos numéricos revela varias ideas clave sobre las características de los diamantes:

Carat (Peso): Los diamantes en el conjunto de datos varían en peso desde 0.2 hasta 5.01 quilates, con un peso promedio de 0.798 quilates. La desviación estándar de 0.474 indica una dispersión considerable en los pesos de los diamantes.

Depth (Profundidad): La profundidad porcentual total de los diamantes oscila entre el 43% y el 79%, con un promedio del 61.75%. La desviación estándar de 1.43 sugiere una variación moderada en la profundidad.

Table (Ancho de la Tabla): El ancho de la tabla varía del 43% al 95%, con un promedio del 57.46%. La desviación estándar de 2.23 indica una amplia gama de anchos de tabla en los diamantes.


Price (Precio): Los precios de los diamantes van desde $326 hasta $18,823, con un precio promedio de $3,932.80. La desviación estándar de $3,989.44 revela una gran dispersión en los precios, lo que sugiere una amplia variedad de calidades y características en los diamantes.

Dimensiones (x, y, z): Las dimensiones de los diamantes (largo, ancho y profundidad) también muestran una variación considerable. Es interesante notar que la dimensión 'y' (ancho) tiene un valor máximo atípico de 58.9 mm, lo que podría indicar un error de datos o un diamante con una forma inusual.


In [26]:
df.dtypes

Unnamed: 0      int64
carat         float64
cut            object
color          object
clarity        object
depth         float64
table         float64
price           int64
x             float64
y             float64
z             float64
dtype: object

Variables Numéricas: Las columnas carat, depth, table, price, x, y, y z son numéricas (float64 para valores decimales e int64 para valores enteros). Estas variables se pueden utilizar directamente para cálculos y análisis estadísticos.
Variables Categóricas: Las columnas cut, color, y clarity son de tipo objeto, lo que sugiere que contienen valores categóricos o de texto. Estas variables requerirán codificación o transformación antes de poder ser utilizadas en modelos de aprendizaje automático.

In [70]:
columns= ['carat', 'depth', 'table', 'price', 'x', 'y', 'z'] 

filtered_df = df[columns]

mean = filtered_df.mean().reset_index().round(3)
median = filtered_df.median().reset_index()
mode = filtered_df.mode().iloc[0].reset_index()

# Renombrar columnas
mean = mean.rename(columns={'index':'Variable',0:'Valor'})
median = median.rename(columns={'index':'Variable',0:'Valor'})
mode = mode.rename(columns={'index':'Variable',0:'Valor'})

mean['Statistic'] = 'Media'
median['Statistic'] = 'Mediana'
mode['Statistic'] = 'Moda'

merged_df = pd.concat([mean, median, mode])

pivoted_df = merged_df.pivot(index='Variable', columns='Statistic', values='Valor').reset_index()

print(pivoted_df.to_markdown(index=True, numalign='left', stralign='center'))


|    |  Variable  | Media   | Mediana   | Moda   |
|:---|:----------:|:--------|:----------|:-------|
| 0  |   carat    | 0.798   | 0.7       | 0.3    |
| 1  |   depth    | 61.749  | 61.8      | 62     |
| 2  |   price    | 3932.8  | 2401      | 605    |
| 3  |   table    | 57.457  | 57        | 56     |
| 4  |     x      | 5.731   | 5.7       | 4.37   |
| 5  |     y      | 5.735   | 5.71      | 4.34   |
| 6  |     z      | 3.539   | 3.53      | 2.7    |


In [63]:
columns= ['carat', 'depth', 'table', 'price', 'x', 'y', 'z']

filtered_df = df[columns]

minimo = filtered_df.min().reset_index()
maximo = filtered_df.max().reset_index()

minimo = minimo.rename(columns={'index':'Variable',0:'Valor'})
maximo = maximo.rename(columns={'index':'Variable',0:'Valor'})

# Add a column called `Statistic` to both `minimo` and `maximo` dataframes
minimo['Statistic'] = 'Minimo'
maximo['Statistic'] = 'Maximo'

# Concatenate the dataframes vertically
merged_df = pd.concat([minimo, maximo])

# Pivot the dataframe to have `Statistic` as columns and `Variable` as the index
pivoted_df = merged_df.pivot(index='Variable', columns='Statistic', values='Valor').reset_index()

# Print the pivoted dataframe in markdown format
print(pivoted_df.to_markdown(index=True, numalign="left", stralign="left"))

|    | Variable   | Maximo   | Minimo   |
|:---|:-----------|:---------|:---------|
| 0  | carat      | 5.01     | 0.2      |
| 1  | depth      | 79       | 43       |
| 2  | price      | 18823    | 326      |
| 3  | table      | 95       | 43       |
| 4  | x          | 10.74    | 0        |
| 5  | y          | 58.9     | 0        |
| 6  | z          | 31.8     | 0        |


In [74]:
import pandas as pd

# Assuming df is already defined
columns = ['carat', 'depth', 'table', 'price', 'x', 'y', 'z']
filtered_df = df[columns]

std = filtered_df.std().reset_index()
quartiles = filtered_df.quantile([0.25, 0.75])
iqr = quartiles.loc[0.75] - quartiles.loc[0.25]

std = std.rename(columns={'index':'Variable', 0:'Valor'})
iqr = iqr.reset_index().rename(columns={'index':'Variable', 0:'Valor'})

std['Statistic'] = 'Desviación Estándar'
iqr['Statistic'] = 'Rango Interquartil'

merged_df = pd.concat([std, iqr])

pivoted_df = merged_df.pivot(index='Variable', columns='Statistic', values='Valor').reset_index()

print(pivoted_df.to_markdown(index=True, numalign="left", stralign="left"))






|    | Variable   | Desviación Estándar   | Rango Interquartil   |
|:---|:-----------|:----------------------|:---------------------|
| 0  | carat      | 0.474011              | 0.64                 |
| 1  | depth      | 1.43262               | 1.5                  |
| 2  | price      | 3989.44               | 4374.25              |
| 3  | table      | 2.23449               | 3                    |
| 4  | x          | 1.12176               | 1.83                 |
| 5  | y          | 1.14213               | 1.82                 |
| 6  | z          | 0.705699              | 1.13                 |
